src/replication/state.cpp

```diff
@@ -222,10 +223,15 @@ bool ReplicationState::HandleVersionMigration(durability::ReplicationRoleEntry &
         main.main_uuid = utils::UUID{};
       }
       data.version = durability::DurabilityVersion::V3;
-      break;
+      [[fallthrough]];
     }
     case durability::DurabilityVersion::V3: {
-      // do nothing - add code if V4 ever happens
+      // Real migration is happening in JSON functions, here just bump version to V4.
```

Comment:
There is no real data change to be made;  but that is not the only thing we need to do here. We are actually updating the durability. So we need to update the durable version as well.